### ISSUE TITLE
Go 1.18 Fuzz tests for Go driver

### ISSUE THREAD

**Issue by protolambda** [ethereum-optimism/optimistic-specs#122]
Go 1.18 introduces fuzzing, neatly integrated into the Go testing framework: https://tip.golang.org/doc/go1.18#fuzzing

We can use it to fuzz the driver (L1 receipt -> L2 deposit).

To improve fuzzing further, speccing "invariants" for the driver would really help: we define a set of rules that can never break. E.g. L1 sender balance decreased by X && mint param is 0, thus L2 must have X extra gas. Etc. etc.

**Comment by maurelian** [comment]
@norswap We're including this in the release candidate?

**Comment by protolambda** [comment]
This was not intended for the deposit rollup release. Go 1.18 fuzzing features are not in a stable Go release yet, expecting that some time in Feb.

**Comment by norswap** [comment]
Yeah I'm not sure why I change the milestone. I think we were discussing it during the daily and wondering why it didn't show up, or maybe I got the wrong issue. Anyhow, no need for RC.
